socorro/external/postgresql/setupdb_app.py

```diff
@@ -219,9 +232,48 @@ def generate_fakedata(self, db, fakedata_days):
             """, dict(zip(["one", "two", "three", "four"],
                       list(fakedata.featured_versions))))
 
-    def main(self):
+    def construct_db_url(self, dbname=None, superuser=False):
+        """Takes a URL to connect to Postgres and updates database name
+            or superuser name/password as indicated"""
+        if self.database_url:
+            database_url = self.database_url
+            url = urlparse.urlparse(database_url)
+            database_username = url.username
+            database_password = url.password
+            database_hostname = url.hostname
+            database_port = url.port
+            # Set the self.database_name because it wasn't set
+            # intentionally by --database_name if database_url
+            # is defined
+            self.database_name = url.path[1:]
+            database_name = dbname or self.database_name
+        else:
+            database_username = self.config.get('database_username')
+            database_password = self.config.get('database_password')
+            database_hostname = self.config.get('database_hostname')
+            database_port = self.config.get('database_port')
+            database_name = dbname
+
+        if superuser:
+            database_username = self.config.get('database_superusername')
+            database_password = self.config.get('database_superuserpassword')
+
+        # construct a URL
+        url = 'postgresql://'
+        if database_username:
+            url += '%s' % database_username
+        if database_password:
+            url += ':%s' % database_password
+        url += '@'
```

Comment:
What if `not database_username and not database_password`?
Then you get `postrgesql://@localhost:5432/foo`? Is that a problem?